poetry/installation/executor.py

```diff
@@ -672,30 +673,40 @@ def _download_link(self, operation: Union[Install, Update], link: Link) -> Link:
                 archive = self._chef.prepare(archive)
 
         if package.files:
-            archive_hash = self._validate_archive_hash(archive, package)
+            archive_hashes = self._validate_archive_hash(archive, package)
 
-            self._hashes[package.name] = archive_hash
+            self._hashes[package.name] = archive_hashes[0]
```

Comment:
This is the bit I'm not sure about -- is it worth adding some code here to prefer a particular hash, or are we satisfied with just picking the first member of the set?